.github/workflows/framework-kind.yml

```diff
@@ -0,0 +1,12 @@
+name: Framework
+
+on:
+  pull_request:
+    branches:
+      - main
+      - release-2.[7-9]
```

Comment:
Optional: We could go ahead and make this `release-[0-9]+.[0-9]+` since there are no other release branches here.